File: songs/clojure/src/t06_function.clj
(ns t06-function (:gen-class) (:use my-songs))

(defn extract
  [songs]
  (remove nil? (map #(get % :tags) songs))
)

(defn -main [& args]
    (doseq [tags (extract songs)]
      (println tags))
  )

File: songs/scala/T06Function.scala
import mysongs._

object T06 extends App {
  def tagsList(songs: List[Song])
    : List[List[String]] = songs map (
      song => song.tags match {
        case Some(tags) => tags
        case None => List()
      }
  )

  tagsList(MySongs.songs) map (
    tags => println(tags)
  )
}
